src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -5225,6 +5271,59 @@ private void ValidateBlockingMode()
         // Helper for SendFile implementations
         private static FileStream OpenFile(string name) => string.IsNullOrEmpty(name) ? null : File.OpenRead(name);
 
+        private SocketError UpdateReceiveSocketErrorForCleanedUp(SocketError socketErrorIn, int bytesTransferred)
+        {
+            // We use bytesTransferred for checking CleanedUp.
+            // When there is a SocketError, bytesTransferred is zero.
+            // An interrupted UDP receive on Linux returns SocketError.Success and bytesTransferred zero.
+            if (bytesTransferred == 0 && CleanedUp)
+            {
+                return IsConnectionOriented ? SocketError.ConnectionAborted : SocketError.Interrupted;
+            }
+            else
+            {
+                return socketErrorIn;
+            }
+        }
+
+        private SocketError UpdateSendSocketErrorForCleanedUp(SocketError socketErrorIn)
```

Comment:
Every use of this is:
```C#
errorCode = UpdateSendSocketErrorForCleanedUp(errorCode);
```
Should this instead return void and take a ref SocketError?